libraries/botbuilder-storage-mssql/src/mssqlStorage.ts

```diff
@@ -0,0 +1,105 @@
+import { Storage, StoreItem, StoreItems } from 'botbuilder-core';
+import { config as msconfig, ConnectionPool, NVarChar, IResult, Int } from 'mssql';
+import { MSSQLOptions } from './schema';
+
+/**
+ * @module @botbuildercommunity/storage-mssql
+ */
+
+function getConnectionPool(connection): ConnectionPool {
+    try {
+        return new ConnectionPool(connection);
+    }
+    catch(e) {
+        console.error(e);
+    }
+}
+
+export class MSSQLStorage implements Storage {
+    private options: MSSQLOptions;
+    private connection: msconfig;
+    public constructor(options: MSSQLOptions) {
+        this.options = options;
+        this.connection = {
+            user: options.dbuser,
+            password: options.dbpassword,
+            server: options.dbserver,
+            database: options.db
+        };
+    }
+    public async read(keys: string[]): Promise<StoreItems> { //Should this be an array of numbers?
+        if(keys == null || keys.length === 0) {
+            return Promise.resolve({});
+        }
+        const pool = await getConnectionPool(this.connection).connect();
```

Comment:
This way; will the connection be shared over multiple requests? Instead of creating a new connection every request.

Reply:
The connection pool will get shared as long as it is not closed. Some literature recommends a singleton for the connection itself, and then only awaiting that singleton when you attempt to connect in order to create a request. I've been all over the mssql and tedious docs though, and haven't found consensus.

In a work project, doing the above while not explicitly closing the pool handles at least 157 concurrent connections and produces 4,000 requests in a 10 second period using the autocannon package.